### PR TITLE
Messaging protocol

### PR DESCRIPTION
**Pull request by OmarQurashi868**
Wrote the technical specifications for the proposed messaging protocol to be used on top of web sockets.
All details are written down in the markdown file added in this PR.

[Trello ticket](https://trello.com/c/1urQb2LJ/33-messaging-protocol)

### CODE REVIEW

**Comment by ahmed-okkio** [comment]
I noticed you're using new lines, would be good to document that

File: MessagingProtocol.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+```
+DO SEND
+FROM 74316
+IN 47532
+WITH
+yoo good morning
+how you been?
+```
+> Request to send a message from user with id `74316` in channel with id `47532`, with the message content being:  
+> `yoo good morning`
```
Comment (ahmed-okkio) [inline MessagingProtocol.md:89]: I'm guessing new lines that appear after the WITH are counted within the string?

Comment (OmarQurashi868) [reply MessagingProtocol.md:89]: Yes, everything after the `WITH` counts as part of the message, until EOF.

**Comment by OmarQurashi868** [comment]
> I noticed you're using new lines, would be good to document that

Updated to include relevant details 👍

**Review by ahmed-okkio — DISMISSED** [review]

**Comment by OmarQurashi868** [comment]
This is an update that implements some changes that came in accordance to new information.
## Highlights
- Clients now just need to pass in the username in the `AUTH` request instead of a token
- Removed the `FROM` parameters from requests from the client, this is because the server should save the username for the current websocket connection
- Added a CONNECT verb
- Added a section for server to client responses

**Review by aeltoum — DISMISSED** [review]

**Comment by OmarQurashi868** [comment]
- Removed old references for user-id left by accident
- Switched to using channel names instead of channel IDs

**Review by ahmed-okkio — DISMISSED** [review]

**Comment by OmarQurashi868** [comment]
Updated protocol to accommodate for message timestamps and message history requests

File: MessagingProtocol.md [hunk 2]
```diff
@@ -146,35 +146,35 @@ When the server receives a `REMEMBER` request from a client, the server is expec
 - ### POPULATE: Send multiple messages at once
    > Required params; `WITH <array-of-messages>`
 
-In the `WITH` argument, seperate message requests should start, end, and be seperated using a sequence of 3 dollar signs `$$$` like so:
+In the `WITH` argument, seperate message requests should start, end, and be seperated using a sequence of 3 dollar signs `/*$*/` like so:
```
Comment (ahmed-okkio) [inline MessagingProtocol.md]: Update paragraph

Comment (OmarQurashi868) [reply MessagingProtocol.md]: Oof, good catch. Done

**Review by ahmed-okkio — APPROVED** [review]

**Comment by ahmed-okkio** [comment]
Try to close this pr it's a bit too big now, if any more changes are needed make a new one

**Pull request merged**